src/commands/aksConnectAcrToCluster/connectAcrToCluster.ts

```diff
@@ -0,0 +1,56 @@
+import { commands, window } from "vscode";
+import { IActionContext } from "@microsoft/vscode-azext-utils";
+import { getReadySessionProvider } from "../../auth/azureAuth";
+import { failed, succeeded } from "../utils/errorable";
+import { InitialSelection } from "../../webview-contract/webviewDefinitions/connectAcrToCluster";
+import { ConnectAcrToClusterDataProvider, ConnectAcrToClusterPanel } from "../../panels/ConnectAcrToClusterPanel";
+import { getExtension } from "../utils/host";
+import * as k8s from "vscode-kubernetes-tools-api";
+import { getAksClusterTreeNode } from "../utils/clusters";
+
+export type ConnectAcrToClusterParams = {
+    initialSelection: InitialSelection;
+};
+
+export function launchConnectAcrToClusterCommand(params: ConnectAcrToClusterParams) {
+    commands.executeCommand("aks.connectAcrToCluster", params);
+}
+
+export async function connectAcrToCluster(_context: IActionContext, target: unknown): Promise<void> {
+    const cloudExplorer = await k8s.extension.cloudExplorer.v1;
+    const params = getConnectAcrToClusterParams(cloudExplorer, target);
+
+    const extension = getExtension();
+    if (failed(extension)) {
+        window.showErrorMessage(extension.error);
+        return;
+    }
+
+    const sessionProvider = await getReadySessionProvider();
+    if (failed(sessionProvider)) {
+        window.showErrorMessage(sessionProvider.error);
+        return;
+    }
+
+    const panel = new ConnectAcrToClusterPanel(extension.result.extensionUri);
+    const dataProvider = new ConnectAcrToClusterDataProvider(sessionProvider.result, params?.initialSelection || {});
```

Comment:
The problem here is the type definition for `ConnectAcrToClusterParams`. The `initialSelection` property should be allowed to be `undefined`, because we can't control how it will be invoked.

I'll fix that, and I'll also separate out the `params?.initialSelection || {}` part into a separate assignment, with a comment that we're explicitly setting an empty `initialSelection` value if it's not defined in the parameters.